client/client.go

```diff
@@ -38,32 +38,42 @@ type DataExportClient interface {
 
 // Client represents a HTTP client for making requests to the FullStory API.
 type Client struct {
-	HTTPClient *http.Client
-	Config     *config.Config
+	httpClient    *http.Client
+	configuration *config.Config
+	authFunc      func(*http.Request,*config.Config)
```

Comment:
I think you can delete everything related to auth. After #92, the only necessary change is making `NewHauser` exported (e.g. moving it out of internal package).